src/lib/envelope/envelope.ml

```diff
@@ -40,6 +47,9 @@ module Incoming = struct
       end
 
       include T
+
+      let equal ~equal_data t t' =
```

Comment:
Should this have a different name, since in general, we have `equal : t -> t -> bool` as a signature.

Reply:
I was getting an error which I misunderstood. I just had to add the dependency in the dune file. Will use the ppx_derive